src/physics/em/detail/TsaiUrbanDistribution.i.hh

```diff
@@ -0,0 +1,49 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2021 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file TsaiUrbanDistribution.i.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "base/Algorithms.hh"
+#include "random/distributions/BernoulliDistribution.hh"
+
+namespace celeritas
+{
+namespace detail
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Construct from input data.
+ */
+CELER_FUNCTION
+TsaiUrbanDistribution::TsaiUrbanDistribution(Energy energy, Mass mass)
+    : energy_(energy), mass_(mass)
+{
```

Comment:
- Note that this should be *inverse* mass like mentioned above.
- Also might be good to add assertions for positivity.
- You also can get away with precalculating and storing `umax_` instead of `energy_` and `mass_`. I suspect the compiler will  optimize it out to the same thing, but it's a cleaner way of separating the "nonrandom" aspects of a distribution from an a particular *sample* of that distribution.